develop/reference_workarounds.rst

```diff
@@ -35,32 +35,21 @@ Citus supports all SQL statements in the multi-tenant use-case. Even in the real
 JOIN a local and a distributed table
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Attempting to execute a JOIN between a local table "local" and a distributed table "dist" causes an error:
+This error no longer occurs in the current version of Citus unless `citus.local_table_join_policy` GUC is set to `never`.
```

Comment:
We should give a reference to the usage of this GUC after we add document it (cc: @SaitTalhaNisanci)